x-pack/plugins/ml/public/application/components/import_export_jobs/import_jobs_flyout/utils.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { JobType } from '../../../../../common/types/saved_objects';
+import type { Job, Datafeed } from '../../../../../common/types/anomaly_detection_jobs';
+import type { DataFrameAnalyticsConfig } from '../../../data_frame_analytics/common';
+
+export interface ImportedAdJob {
+  job: Job;
+  datafeed: Datafeed;
+}
+
+export interface JobIdWrapper {
+  id: string;
+  originalId: string;
+  valid: boolean;
+  invalidMessage: string;
+}
+
+export interface SkippedJobs {
+  jobId: string;
+  missingIndices: string[];
+}
+
+function isImportedAdJobs(obj: any): obj is ImportedAdJob[] {
+  return Array.isArray(obj) && obj.some((o) => o.job && o.datafeed);
+}
+
+function isDataFrameAnalyticsConfigs(obj: any): obj is DataFrameAnalyticsConfig[] {
+  return Array.isArray(obj) && obj.some((o) => o.dest && o.analysis);
+}
+
+function readFile(file: File) {
+  return new Promise((resolve, reject) => {
+    if (file && file.size) {
+      const reader = new FileReader();
+      reader.readAsText(file);
+
+      reader.onload = (() => {
+        return () => {
+          const data = reader.result;
+          if (typeof data === 'string') {
+            try {
+              const json = JSON.parse(data);
+              resolve(json);
+            } catch (error) {
+              reject();
+            }
+          } else {
+            reject();
+          }
+        };
+      })();
+    } else {
+      reject();
+    }
+  });
+}
+export async function readJobConfigs(
```

Comment:
can all these util functions be combined into a service?  For instance `JobsExportService ` I mentioned above

Reply:
updated here https://github.com/elastic/kibana/pull/101037/commits/b42b9025425bab93bcc9c770ff16e3ac9be27dca